haiku/_src/conv_test.py

```diff
@@ -340,6 +385,26 @@ def f():
     out = np.squeeze(out, axis=(0, 4))
     np.testing.assert_equal(out, expected_out)
 
+  def test_invalid_input_shape(self):
```

Comment:
nit: It would be easier to express this case using `@test_util.transform_and_run`. then you don't need to use base.transform and create an inner function :smile:. Same for the functions below.

Reply:
Got it! Let me fix those using `@test_util.transform_and_run` :)

Reply:
Done! @tomhennigan